numpyro/distributions/distribution.py

```diff
@@ -155,9 +158,13 @@ def _validate_sample(self, value):
 
     def __call__(self, *args, **kwargs):
```

Comment:
Refer to my comment below regarding keeping this method as is and instead using `transform_with_intermediates` in the handler directly.